src/DepthImageToLaserScan.cpp

```diff
@@ -115,8 +115,12 @@ sensor_msgs::LaserScanPtr DepthImageToLaserScan::convert_msg(const sensor_msgs::
   scan_msg->range_min = range_min_;
   scan_msg->range_max = range_max_;
   
+  double center_row = cam_model_.cy()*2*scan_offset_;
+  double bottom_row = center_row - scan_height_/2;
+  double top_row = center_row + scan_height_/2;
+
   // Check scan_height vs image_height
-  if(scan_height_/2 > cam_model_.cy() || scan_height_/2 > depth_msg->height - cam_model_.cy()){
+  if(bottom_row < 0 || top_row > depth_msg->height){
```

Comment:
Do you mean like this, Dr. Damien?
@damienjadeduff

Reply:
I think so.

Reply:
But there is still redundancy. Lines 123 and 130 express the same condition.

Reply:
Sorry, I was looking at the old version.

Reply:
It seems fine.